code/src/cljs/sixsq/nuvla/ui/edge_detail/views.cljs

```diff
@@ -1460,6 +1584,229 @@
+                        :on-change #(do
+                                      (swap! form-data assoc :enabled (not (:enabled @form-data))))}]]
+            [ui/FormField {:label "Run"
+                           :required  true}]
+            "Shell script: "
+            [uix/EditorShell
+             ""
+             (fn [_editor _data value]
+               (ui-callback/input-callback
+                 (swap! form-data assoc :run value)))
+             true
+             "full-width"]]]]
+         [ui/ModalActions
+          [uix/Button
+           {:text     (@tr [:add])
+            :disabled (utils/form-add-playbook-incomplete? @form-data)
+            :primary  true
+            :on-click on-click-fn}]]]))))
+
+
+(defn TabPlaybooks
+  []
+  (let [tr             (subscribe [::i18n-subs/tr])
+        nuvlabox       (subscribe [::subs/nuvlabox])
+        playbooks      (subscribe [::subs/nuvlabox-playbooks])
+        selected-playbook (subscribe [::subs/nuvlabox-current-playbook])
+        run-changed?   (r/atom false)
+        run            (r/atom nil)
+        enabled-changed?  (r/atom false)]
+    (fn []
+      (let [n     (count @playbooks)
+            on-change-fn   (fn [k]
+                             (dispatch [::events/get-nuvlabox-current-playbook k]))]
+        [ui/TabPane
+         (when (nil? (:host-level-management-api-key @nuvlabox))
+           [ui/Message {:warning true} (@tr [:nuvlabox-playbooks-disabled])])
+
+         [ui/Container {:text-align "center"}
+          [ui/Label {:basic true}
+           "Total NuvlaBox playbooks found: "
+           [ui/LabelDetail n]]
+
+          [ui/Segment {:secondary true
+                       :color     "grey"
+                       :raised    true}
+
+           [ui/Container
+            [ui/Dropdown
+             {:header      (r/as-element [ui/DropdownHeader {:content "Select Playbook"
+                                                             :icon "book"}])
+              :placeholder (str (@tr [:select]) " playbook")
+              :on-change   (ui-callback/value on-change-fn)
+              :selection   true
+              :search      true
+              ;:icon        ""
+              :style       {:margin "1em"}
+              :clearable   false
+              :options     (map (fn [{:keys [id name type enabled]}]
+                                  ^{:key id}
+                                  {:key id,
+                                   :text (or name id),
+                                   :value id,
+                                   :description  (if enabled
+                                                   "enabled"
+                                                   "disabled")
+                                   :icon (if (= "EMERGENCY" type)
+                                           "emergency"
+                                           "wrench")}) @playbooks)}]
+            [ui/Button {:icon "plus"
```

Comment:
There is no check if the user has enough rights to be able to add a playbook.

Reply:
check added